internal/config/analytics.go

```diff
@@ -60,8 +60,9 @@ func (c *ClickhouseConfig) Options() (*clickhouse.Options, error) {
 
 // PrometheusConfig defines the connection details for connecting Flipt to Prometheus.
 type PrometheusConfig struct {
-	Enabled bool   `json:"enabled,omitempty" mapstructure:"enabled" yaml:"enabled,omitempty"`
-	URL     string `json:"url,omitempty" mapstructure:"url" yaml:"url,omitempty"`
+	Enabled   bool   `json:"enabled,omitempty" mapstructure:"enabled" yaml:"enabled,omitempty"`
+	URL       string `json:"-" mapstructure:"url" yaml:"-"`
+	AuthToken string `json:"-" mapstructure:"auth_token" yaml:"-"`
```

Comment:
what if instead we make `Authentication` an object, and have `Token` a field under it?

Im a bit worried that this is Fly.io specific, because Prometheus supports basic username/pw authentication per its docs, not token auth

Reply:
I agree with you that from usability prospective it could be more user-friendly. But with Prometheus the auth token will be `Basic base64(user:pass)` for basic username/password.  I have updated the example and added auth for Prometheus in this PR and it works fine. For Fly.io it could be two options `Bearer longtoken` or `FlyV1 fm2_lJPECAAAAAAAAC...`.  And there are other Prometheus compatible services which could have some other options.

I really don't want to create a complex auth configuration object for it right now without good insights. Could I mark is as experimental and wait if any feedback appears about this feature? wdyt?

Reply:
Yeah, let's do that